src/irr/asset/CXMeshFileLoader.cpp

```diff
@@ -225,12 +225,12 @@ core::matrix4x3 getGlobalMatrix_evil(asset::ICPUSkinnedMesh::SJoint* joint)
 }
 
 
-bool CXMeshFileLoader::load(SContext& _ctx, io::IReadFile* file)
+bool CXMeshFileLoader::load(SContext& _ctx, io::IReadFile* file, const asset::IAssetLoader::SAssetLoadParams& _params)
```

Comment:
@devshgraphicsprogramming  should _params be a const& member or should it be passed through functions?

Reply:
and which one do you think will crash if you attempt to call the loader from multiple threads?

Reply:
being honest I don't know, I haven't done anything with multithreading before

Reply:
well if you have a member and the loader is shared between threads, what's gonna happen?

Reply:
I think that operations on such a member won't be atomic, right? So it could crash

Reply:
well done, now you know why every loader has a context struct and passes a reference to it as a parameter.... rule nr. 1 of multi-threading, avoid global states.